pkg/sdk/testint/secrets_gen_integration_test.go

```diff
@@ -0,0 +1,673 @@
+		require.Error(t, err)
+	})
+
+	t.Run("Drop: non-existing", func(t *testing.T) {
+		id := testClientHelper().Ids.RandomSchemaObjectIdentifier()
+
+		err := client.Secrets.Drop(ctx, sdk.NewDropSecretRequest(id))
+		assert.ErrorIs(t, err, sdk.ErrObjectNotExistOrAuthorized)
+	})
+
+	t.Run("Show", func(t *testing.T) {
+		secretOAuthClientCredentials, _ := createSecretWithOAuthClientCredentialsFlow(t, integrationId, []sdk.SecurityIntegrationScope{{Scope: "foo"}}, nil)
+		secretOAuthAuthorizationCode, _ := createSecretWithOAuthAuthorizationCodeFlow(t, integrationId, "foo", refreshTokenExpiryTime, nil)
+		secretBasicAuthentication, _ := createSecretWithBasicAuthentication(t, "foo", "bar", nil)
+		secretGenericString, _ := createSecretWithGenericString(t, "foo", nil)
+
+		returnedSecrets, err := client.Secrets.Show(ctx, sdk.NewShowSecretRequest())
+		require.NoError(t, err)
+		require.Contains(t, returnedSecrets, *secretOAuthClientCredentials)
+		require.Contains(t, returnedSecrets, *secretOAuthAuthorizationCode)
+		require.Contains(t, returnedSecrets, *secretBasicAuthentication)
+		require.Contains(t, returnedSecrets, *secretGenericString)
+	})
+
+	t.Run("Show: SecretWithOAuthClientCredentialsFlow with Like", func(t *testing.T) {
+		secret1, id1 := createSecretWithOAuthClientCredentialsFlow(t, integrationId, []sdk.SecurityIntegrationScope{{Scope: "foo"}}, nil)
+		secret2, _ := createSecretWithOAuthClientCredentialsFlow(t, integrationId, []sdk.SecurityIntegrationScope{{Scope: "bar"}}, nil)
+
+		returnedSecrets, err := client.Secrets.Show(ctx, sdk.NewShowSecretRequest().WithLike(sdk.Like{
+			Pattern: sdk.Pointer(id1.Name()),
+		}))
+		require.NoError(t, err)
+		require.Contains(t, returnedSecrets, *secret1)
+		require.NotContains(t, returnedSecrets, *secret2)
+	})
+
+	t.Run("Show: SecretWithOAuthAuthorization with Like", func(t *testing.T) {
+		secret1, id1 := createSecretWithOAuthAuthorizationCodeFlow(t, integrationId, "foo_1", refreshTokenExpiryTime, nil)
+		secret2, _ := createSecretWithOAuthAuthorizationCodeFlow(t, integrationId, "foo_2", refreshTokenExpiryTime, nil)
+
+		returnedSecrets, err := client.Secrets.Show(ctx, sdk.NewShowSecretRequest().WithLike(sdk.Like{
+			Pattern: sdk.Pointer(id1.Name()),
+		}))
+		require.NoError(t, err)
+		require.Contains(t, returnedSecrets, *secret1)
+		require.NotContains(t, returnedSecrets, *secret2)
+	})
+
+	t.Run("Show: SecretWithBasicAuthentication with Like", func(t *testing.T) {
+		secret1, id1 := createSecretWithBasicAuthentication(t, "foo_1", "bar_1", nil)
+		secret2, _ := createSecretWithBasicAuthentication(t, "foo_2", "bar_2", nil)
+
+		returnedSecrets, err := client.Secrets.Show(ctx, sdk.NewShowSecretRequest().WithLike(sdk.Like{
+			Pattern: sdk.Pointer(id1.Name()),
+		}))
+		require.NoError(t, err)
+		require.Contains(t, returnedSecrets, *secret1)
+		require.NotContains(t, returnedSecrets, *secret2)
+	})
+
+	t.Run("Show: SecretWithGenericString with Like", func(t *testing.T) {
```

Comment:
I wouldn't go that deep to test SHOW LIKE and SHOW IN for every secret type, but if we already have it lets leave it.